src/pages/index.tsx

```diff
@@ -40,8 +40,34 @@ const IndexPage: React.FC<IndexPageProps> = ({ location }) => {
       <BlockListComponent variables={{ offset, limit }}>
         {({ data, loading, error }) => {
           if (error) return <p>error!</p>;
+
+          const timestamps: Date[] | null = data && data.blocks
+            ? data.blocks.map(block => new Date(block!.timestamp))
+            : null;
+
+          let interval : number | null = timestamps ? 0 : null;
+          if (interval != null && timestamps) {
+            for (let i = 0; i < timestamps.length - 1; i++) {
+              interval += +timestamps[i] - +timestamps[i + 1];
+            }
+            interval /= (timestamps.length - 1) * 1000;
+          }
+
+          const difficulties: number[] | null = data && data.blocks
+            ? data.blocks.map(block => block!.difficulty)
+            : null;
+          let difficulty = 0;
+          if (difficulty != null && difficulties) {
+            for (let i = 0; i < difficulties.length; i++)
```

Comment:
We can also use [`Array.prototype.reduce()`](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce).

```suggestion
            difficulty = difficulties.reduce((d, sum) => d + sum, 0);
```